test/agents/test_role_playing.py

```diff
@@ -146,3 +146,33 @@ def test_role_playing_with_function():
         assert isinstance(response.terminated, bool)
         assert response.terminated is False
         assert isinstance(response.info, dict)
+
+
+def test_role_playing_role_sequence(model_type=None):
+    task_prompt = "Develop a trading bot for the stock market"
+    role_playing = RolePlaying(
+        assistant_role_name="Python Programmer",
+        assistant_agent_kwargs=dict(model=model_type),
+        user_role_name="Stock Trader",
+        user_agent_kwargs=dict(model=model_type),
+        task_prompt=task_prompt,
+        with_task_specify=True,
+        task_specify_agent_kwargs=dict(model=model_type),
+    )
+    assistant_role_sequence = []
+    user_role_sequence = []
+
+    input_assistant_msg, _ = role_playing.init_chat()
+    assistant_response, user_response = role_playing.step(input_assistant_msg)
+    input_assistant_msg = assistant_response.msg
+    assistant_response, user_response = role_playing.step(input_assistant_msg)
+
+    for record in role_playing.user_agent.stored_messages:
+        user_role_sequence.append(record.role_at_backend)
+    for record in role_playing.assistant_agent.stored_messages:
+        assistant_role_sequence.append(record.role_at_backend)
+
+    assert user_role_sequence == \
+           ['system', 'assistant', 'user', 'assistant', 'user']
+    assert assistant_role_sequence == \
```

Comment:
This must be ['system', 'user', 'assistant', 'user', 'assistant', ...] as well

Reply:
In init chat, the assistant agent takes one step which add a 'user' in the list, but it didn't have submmit excute, so there is a missing 'assistant' in this list, or have one more 'user'.

Reply:
@zhiyu-01 Thank you for finding this out. It is indeed looks like a bug, but it is not critical, so we won't fix it now. @lightaime tests like this were important to have long ago. Consider opening a bug ticker if you think that
```
['system', 'user', 'user', 'assistant', 'user', 'assistant']
```
is not intended and must be fixed to eliminate two 'user' records.

Reply:
It is intended to drop the initial message from the assistant agent. But it is also fine to submit it. If it is too confusing.